addons/website/models/website.py

```diff
@@ -261,10 +261,10 @@ def _handle_domain(self, vals):
             vals['domain'] = vals['domain'].rstrip('/')
 
     @api.ondelete(at_uninstall=False)
-    def _unlink_except_last_remaining_website(self):
-        website = self.search([('id', 'not in', self.ids)], limit=1)
-        if not website:
-            raise UserError(_('You must keep at least one website.'))
+    def _unlink_except_default_website(self):
+        default_website = self.env.ref('website.default_website', raise_if_not_found=False)
+        if default_website and default_website in self:
+            raise UserError(_("You cannot delete default website %s. Try to change its settings instead", default_website.name))
```

Comment:
I'll let you decide what you feel more confortable with here, but just to be sure that you know/realize:
This will break every translations, meaning that a DB let's say in a company in Arab (to illustrate with the best example) which have no clue about english, they will now see the English message as it will take time to be re-translated again (probably never in some language actually).

So here it's not a tradeoff I usually accept in our team: we polish the english wording (very low value) but we break the translation (very high risk).

Bonus point, you won't even need to do anything in the `.pot` in your master forward port where you will change the string as we don't do anything about `.pot` (maybe except removal) in master :+1: (it's auto generated when we freeze the next version)

Reply:
@sauc-odoo 
I think you should keep original check ('You must keep at least one website.') and then add the new check.
Protecting  default website from deleting assumes that there is at least one website, but we must keep in mind, that in some databases the default website is already deleted.

Reply:
It really looks weird to go with the `.` at the end of sentence way but not for the second sentence, but let's not nitpicking on this one any longer.